java/form/src/org/netbeans/modules/form/FormLoaderSettings.java

```diff
@@ -472,12 +472,13 @@ public void setPaletteInToolBar(boolean value) {
      * @return background color of the designer.
      */
     public java.awt.Color getFormDesignerBackgroundColor() {
-        Color defaultBackground = UIManager.getColor( "Tree.background" ); //NOI18N
-        if( null == defaultBackground )
+        Color defaultBackground = UIManager.getColor("nb.formdesigner.designer.background"); //NOI18N
+        if (defaultBackground == null)
+            defaultBackground = UIManager.getColor("Tree.background"); //NOI18N
+        if (defaultBackground == null)
```

Comment:
Which LAF is used in the form designer? Is it the same as the IDE? In that case, should the background color not be taken from the LAF rather than from a special constant? (Also, I wonder why "Tree.background" was used previously instead of the default JPanel background.)

Reply:
As far as I know the form designer always uses the IDE LAF.
But it is possible to preview the form with other LAFs in new window. (right-click on panel: "Preview Design > LAF")

> In that case, should the background color not be taken from the LAF rather than from a special constant?

No sure what you mean. `UIManager.getColor()` gets the color from the LAF (if set).

`Tree.background` was probably used because it is white, while `Panel.background` is gray.
BTW `nb.formdesigner.designer.background` is currently not used. I've added it for completeness.

Reply:
Oh, right, the constant is from the LAF. My main concern was that it would seem logical to take it from the JPanel background, rather than invent a new constant. But since the existing code doesn't do that, I suppose it makes sense to add a constant to preserve backwards-compatibility.

How come Tree.background is not dark for FlatLAF, though?

Reply:
> How come Tree.background is not dark for FlatLAF, though?

It is dark and it is used.

I've first added UI value for the designer border color (because it was too bright) and thought that it is good idea to do the same for the designer background in case need to change it.

Reply:
Sounds reasonable.